semgrep-core/tests/OTHER/rules/taint_labels2.yaml

```diff
@@ -0,0 +1,17 @@
+rules:
+  - id: tainting
+    mode: taint
+    languages:
+      - python
+    message: |
+      This confirms taint mode works.
+    pattern-sinks:
+      - requires: TAINTED and not CLEANED
```

Comment:
Does that mean sanitizer are now useless? Or they are still useful as a quicker way to express
source/sanitize/sink instead of having to invent labels and requires expressions?

Reply:
I am not sure to be honest, I think we'll have to gain some experience in using labels. Sanitizers do allow you to _completely remove_ taint from data, although I think that can be simulated with labels too. I could have translated (internally) `pattern-sanitizers` to `patter-sources` with a `"__SANITIZED__"` label, and then rewrite the `requires` of sinks as `(...) not __SANITIZED__`. But `not-conflicting` sanitizers cannot be simulated that way. Although I think we should replace `not-conflicting` with some `options:` because they are a bit confusing to use sometimes.